core/lib/zksync_core/src/native_erc20_fetcher/mod.rs

```diff
@@ -0,0 +1,107 @@
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use metrics::atomics::AtomicU64;
+use tokio::{
+    sync::{watch, OnceCell},
+    task::JoinHandle,
+};
+use zksync_config::configs::native_erc20_fetcher::NativeErc20FetcherConfig;
+
+// TODO: this error type is also defined by the gasAdjuster module,
+//       we should consider moving it to a common place
+#[derive(thiserror::Error, Debug, Clone)]
+#[error(transparent)]
+pub struct Error(Arc<anyhow::Error>);
+
+impl From<anyhow::Error> for Error {
+    fn from(err: anyhow::Error) -> Self {
+        Self(Arc::new(err))
+    }
+}
+
+pub trait Erc20Fetcher: 'static + std::fmt::Debug + Send + Sync {
+    fn conversion_rate(&self) -> anyhow::Result<u64>;
```

Comment:
Yeah, I think we'll have to use some decimal crate and express the conversion rate as one. This would also be more in line with apis like Coingecko that return conversion rates as floats/fractional numbers. My only concern is what happens on overflow, as the gas price is currently a `u64` (discussed [here](https://github.com/matter-labs/zksync-era/pull/996#discussion_r1504827448) as well)

Reply:
Yup, using `U256` is a prerequisite.
By the way. Given that zkSync Lite supports paying fee in any token, it already has an infrastructure for converting tokens. You can have a look at the [`fee_ticker` module in Lite repo](https://github.com/matter-labs/zksync/blob/master/core/bin/zksync_api/src/fee_ticker/mod.rs) -- maybe you'll be able to hijack some stuff from there.

Reply:
Awesome, thanks for the link! Tracked the change to use decimals on the conversion rate [here](https://github.com/lambdaclass/zksync-era/issues/201)